src/actions/query-keyspaces.ts

```diff
@@ -3,23 +3,38 @@
 import { actionClient } from "@scylla-studio/lib/safe-actions";
 import { z } from "zod";
 
-import { BatchStatement, Cluster, Query } from "@lambda-group/scylladb";
+import { Cluster } from "@lambda-group/scylladb";
 import { parseKeyspaces } from "@scylla-studio/lib/cql-parser/parser";
 
 
 
 export const queryKeyspaceAction = actionClient
-  .schema(z.object({}))
-  .action(async ({ parsedInput }) => {
-    const cluster = new Cluster({
-      nodes: ["localhost:9042"],
+    .schema(
+        z.object({
+            host: z.string().min(1, "Host is required"),
+            port: z.number().min(1, "Port is required"),
+            username: z.string().nullable(),
+            password: z.string().nullable(),
+        })
+    )
+    .action(async ({ parsedInput }) => {
+        const { host, port, username, password } = parsedInput;
+
+        const clusterConfig: any = {
+            nodes: [`${host}:${port}`],
+        };
+
+        if (username && password) {
+            clusterConfig.auth = {
+                username,
+                password,
+            };
+        }
```

Comment:
You might as well check the [src/actions/execute-query.ts](https://github.com/basementdevs/scylla-studio/pull/22/files#diff-a624c47dd8ce7ac4b51e54c369ef4af92cf8b3e24a8b45c29ae1e594eb707f81) file to see how the auth is being established and do roughly the same.

Reply:
ok